src/transport/SessionManager.h

```diff
@@ -176,6 +176,10 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate
     void ExpireAllPairingsForFabric(FabricIndex fabric);
     void ExpireAllPASEPairings();
 
+    // This API is used by UpdateNOC command, to invalidate all sessions except the given one, whose release is deferred until
+    // UpdateNOC command finishing its work.
+    void ReleaseSessionsForFabricExceptOne(FabricIndex fabricIndex, const SessionHandle & deferred);
```

Comment:
I think this should just take a SecureSession in some form for the `deferred` arg, instead of assuming that's what gets passed in.

Reply:
This one is call by `ExchangeManager::AbortExchangesForFabricExceptOne`, generally we should avoid using raw session pointer outside transport layer.

But I can add constraint to this function.